src/ast/attachpoint_parser.cpp

```diff
@@ -288,6 +290,22 @@ AttachPointParser::State AttachPointParser::special_parser()
   return OK;
 }
 
+AttachPointParser::State AttachPointParser::benchmark_parser()
```

Comment:
hmm, maybe this should instead go in the `special_parser`

Reply:
This is where I started but ended up separating them to avoid conditionals special-casing "BENCH" probes through the stack. The two key differences are parsing out the benchmark name and using `BPF_PROG_TYPE_XDP` instead of `BPF_PROG_TYPE_RAW_TRACEPOINT`.

Reply:
This is where I started but ended up separating them to avoid conditionals special-casing "BENCH" probes through the stack. The two key differences are parsing out the benchmark name and using `BPF_PROG_TYPE_XDP` instead of `BPF_PROG_TYPE_RAW_TRACEPOINT`.